modules/checklist/steps/step-base.php

```diff
@@ -191,4 +196,36 @@ public function get_step_initial_progress() : array {
 	private function set_step_progress() : void {
 		$this->module->set_step_progress( $this->get_id(), $this->user_progress );
 	}
+
+	public function page_exists( $params ) : bool {
```

Comment:
All these new methods should be protected as they will only be used by the steps, and they should be added after the last protected method, before the first private method